tests/MongoDB.EntityFrameworkCore.UnitTests/Extensions/MongoPropertiesConfigurationBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using MongoDB.Bson;
+
+namespace MongoDB.EntityFrameworkCore.UnitTests.Extensions;
+
+public class MongoPropertiesConfigurationBuilderExtensionsTests
+{
+    [Theory]
+    [InlineData(BsonType.String)]
+    [InlineData(BsonType.Double)]
+    [InlineData(BsonType.Decimal128)]
+    [InlineData(BsonType.Int64)]
+    public void HaveBsonRepresentation_can_set_type_on_property(BsonType bsonType)
+    {
+        using var db = new TestDbContext(bsonType);
+
+        var property = db.GetProperty((TestEntity t) => t.DateTimeOffsetProperty);
+        Assert.NotNull(property);
+
+        var representation = property.GetBsonRepresentation();
+        Assert.NotNull(representation);
```

Comment:
Thanks for the PR, looks great, just missing the Assert.Equal(bsonType, representation.BsonType) line here.

Reply:
Thanks. Fixed it